lib/Frontend/DiagnosticVerifier.cpp

```diff
@@ -571,18 +625,18 @@ DiagnosticVerifier::verifyFile(unsigned BufferID, bool shouldAutoApplyFixes) {
 
     if (I == CapturedDiagnostics.end()) continue;
 
-    if (I->getMessage().find(expected.MessageStr) == StringRef::npos) {
+    if (I->Message.find(expected.MessageStr) == StringRef::npos) {
       auto StartLoc = SMLoc::getFromPointer(expected.MessageRange.begin());
       auto EndLoc = SMLoc::getFromPointer(expected.MessageRange.end());
 
-      llvm::SMFixIt fixIt(llvm::SMRange{StartLoc, EndLoc}, I->getMessage());
+      llvm::SMFixIt fixIt(llvm::SMRange{StartLoc, EndLoc}, I->Message);
       addError(expected.MessageRange.begin(), "incorrect message found", fixIt);
-    } else if (I->getColumnNo() + 1 != (int)*expected.ColumnNo) {
+    } else if (I->Column != *expected.ColumnNo) {
```

Comment:
Were we off-by one?

Reply:
This is just a consequence of operating on `DiagnosticInfo` instead of `SMDiagnostic`. I don't know why their column indexing is inconsistent, but I'm hesitant to change the behavior as part of this PR